### DISCUSSION TITLE
Control bending direction of IK? (Pole targets)

### DISCUSSION THREAD

**Discussion by Silverlan** [bepu/bepuphysics1#52] [General]
I have the following problem:

I have a simple IK setup similar to the [action figure in the IK demo](https://github.com/bepu/bepuphysics1/blob/88243f8a688b792d4f5b617c67fce7847b2f78f0/BEPUphysicsDemos/Demos/Extras/Tests/InverseKinematicsTestDemo.cs#L381). My character has drag controls for the hands and for the feet. In general it works quite well, but in some cases the arms or legs don't quite bend the way I want. I would like to add an additional control for the knee/elbow, to allow me to adjust the bend direction in such cases.

An additional drag control for the knee would accomplish that but that will cause the entire ik to be dragged towards the knee, including the foot. I need to be able to change the knee's bending direction without affecting the foot's position.

In Blender/Maya/Unreal, you can use Pole Targets/Vectors ( https://artisticrender.com/what-is-a-pole-target-and-how-do-we-use-it-in-blender/ ) for this. Is there something similar in BEPUik that I've maybe missed? If not, any tips on how something like this could be implemented? I'm not entirely sure how to go about it.

Thank you!

**Comment by RossNordby** [comment]
`IKSwingLimit` could be used to restrict the bend to only the desired side. It doesn't proactively influence the bend direction, so there's still a risk of getting 'trapped' in a local minimum if the knee/elbow still inverts far enough that the solver thinks it's the best it can do, but it typically works alright. If there are issues, forcing a very slight bend alongside the swing limit is typically sufficient to resolve them.

**Comment by Silverlan** [comment]
> `IKSwingLimit` could be used to restrict the bend to only the desired side. It doesn't proactively influence the bend direction, so there's still a risk of getting 'trapped' in a local minimum if the knee/elbow still inverts far enough that the solver thinks it's the best it can do, but it typically works alright. If there are issues, forcing a very slight bend alongside the swing limit is typically sufficient to resolve them.

Hm... I'm looking for a method that's a bit more user-controllable. I think I may be able to re-purpose the revolute control for my purposes by making it align the bone (e.g. the knee) with the direction to the control point instead of using a world-axis. I noticed, however, that the _Rigidity_ doesn't seem to have any effect on the revolute control, even when set to very low values (e.g. 0.0001). Is there another way to change how much influence the control has on the IK?

**Reply by RossNordby** [reply]
There's `MaximumForce`, but `Rigidity` *should* have some visible effect. If 0.00001 and 100 exhibit exactly the same behavior, there may be some other configuration wonkiness going on.
